test/unit/test_tv.py

```diff
@@ -0,0 +1,58 @@
+
+
+import reciprocalspaceship as rs
+import gemmi as gm
+import numpy as np
+
+from pytest import fixture
+from meteor import tv
+
+
+@fixture
+def flat_difference_map() -> rs.DataSet:
+    """
+    A simple 3x3x3 P1 map, random
+    """
+
+    params = (10.0, 10.0, 10.0, 90.0, 90.0, 90.0)
+    cell = gm.UnitCell(*params)
+    sg_1 = gm.SpaceGroup(1)
+    Hall = rs.utils.generate_reciprocal_asu(cell, sg_1, 5.0, anomalous=False)
+
+    h, k, l = Hall.T
+    ds = rs.DataSet(
+        {
+            "H": h,
+            "K": k,
+            "L": l,
+            "DF": np.random.randn(len(h)),
+            "PHIC": np.zeros(len(h)),
+        },
+        spacegroup=sg_1,
+        cell=cell,
+    ).infer_mtz_dtypes()
+    ds.set_index(["H", "K", "L"], inplace=True)
+
+    return ds
+
+
+def test_tv_denoise_difference_map_smoke(flat_difference_map: rs.DataSet) -> None:
```

Comment:
What is smoke?

Reply:
[smoke tests](https://en.wikipedia.org/wiki/Smoke_testing_(software)) usually just make sure a function runs and returns the intended kind of output, without ensuring that output is correct